weavertest/internal/simple/simple_test.go

```diff
@@ -80,39 +71,32 @@ func TestTwoComponents(t *testing.T) {
 		]`},
 	} {
 		t.Run(c.name, func(t *testing.T) {
-			file := filepath.Join(t.TempDir(), fmt.Sprintf("simple_%s", uuid.New().String()))
-
-			root := weavertest.Init(ctx, t, weavertest.Options{
+			opts := weavertest.Options{
 				SingleProcess: c.single,
 				Config:        c.config,
-			})
-			src, err := weaver.Get[simple.Source](root)
-			if err != nil {
-				t.Fatal(err)
-			}
-			dst, err := weaver.Get[simple.Destination](root)
-			if err != nil {
-				t.Fatal(err)
 			}
+			weavertest.Run(t, opts, func(src simple.Source, dst simple.Destination) {
+				file := filepath.Join(t.TempDir(), fmt.Sprintf("simple_%s", uuid.New().String()))
+				want := []string{"a", "b", "c", "d", "e"}
+				for _, in := range want {
+					if err := src.Emit(ctx, file, in); err != nil {
+						t.Fatal(err)
+					}
+				}
 
-			want := []string{"a", "b", "c", "d", "e"}
-			for _, in := range want {
-				if err := src.Emit(ctx, file, in); err != nil {
+				got, err := dst.GetAll(ctx, file)
+				if err != nil {
 					t.Fatal(err)
 				}
-			}
-
-			got, err := dst.GetAll(ctx, file)
-			if err != nil {
-				t.Fatal(err)
-			}
-			if !reflect.DeepEqual(want, got) {
-				t.Fatalf("GetAll() = %v; expecting %v", got, want)
-			}
+				if !reflect.DeepEqual(want, got) {
+					t.Fatalf("GetAll() = %v; expecting %v", got, want)
+				}
+			})
 		})
 	}
 }
 
+/*
```

Comment:
Should we delete this test? I'm not sure it's testing much.

Reply:
It is nice to make sure that the Listener() method works. However I don't know how to get a handle to the underlying implementation, which is why it is commented out.

I will leave it commented out with an explanatory comment and maybe we can get back to it soon.